knitr/curse-dims/curse-dims.Rmd

```diff
@@ -0,0 +1,770 @@
+$\mathsf{ChiSquare}(n)$
+    - this limits Hamiltonian Monte Carlo's mixing
+
+The exercises are organized around computation, analytic derivations,
+and generalizations to non-Euclidean distance such as taxicab distance
+and its relation to the Laplace distribution, and Riemannian manifolds
+and their relation to multivariate normal distributionss.
+
+<hr />
+
+
+## 1. Euclidean Length and Distance
+
+### Euclidean Length
+
+Consider a vector $y = (y_1, \ldots, y_N)$ with $N$ elements (we call
+such vectors $N$-vectors).  The Euclidean length of a vector $y$ is
+written as $|| y ||$, and defined by a generalizing the Pythagorean
+theorem,
+
+\[
+|| y || 
+\ = \ \sqrt{y_1^2 + y_2^2 + \cdots + y_N^2}.
+\]
+
+In words, the length of a vector is the square root of the sum of the
+squares of its elements. 
+
+If we take $y = (y_1, \ldots, y_N)$ to be a row vector, then its
+length is determined by a dot product, which can be written using
+matrix notation as
+
+\[
+|| y || = \sqrt{y \, y^{\top}}.
+\]
+
+#### Calculating Vector Length in R
+
+The following function computes vector length in R.
+
+```{r comment=NA}
+euclidean_length <- function(u) sqrt(sum(u * u));
+```
+
+In R, the operator `*` operates elementwise rather than as vector
+multiplication.  To test the function on a simple case, we can verify
+the first example of the Pythagorean theorem everyone learns in
+school, that $|| (3, 4) || = 5$.
+
+```{r comment=NA}
+euclidean_length(c(3, 4));
+```
+
+In R, the function `length()` returns the number of elements in a
+vector rather than the vector's Euclidean length.
+
+```{r comment=NA}
+length(c(3, 4));
+```
+
+### Euclidean Distance
+
+The Euclidean distance between two $N$-vectors, $x = (x_1, \ldots,
+x_N)$ and $y = (y_1, \ldots, y_N)$, written $\mathrm{d}(x,y)$, is the
+Euclidean length of the straight line connecting them.
+
+\[ 
+\mathrm{d}(x, y)
+\ = \
+|| x - y ||
+\ = \
+\sqrt{(x_1 - y_1)^2 + \cdots + (x_N - y_N)^2} 
+\] 
+
+
+## 2.  All the Mass is in the Corners
+
+Suppose we have a square and inscribe a circle in it, or that we have
+a cube and a sphere inscribed in it.  When we extend this construction
+to higher dimensions, we get hyperspheres inscribed in hypercubes.
+This section illustrates the curious fact that as the dimensionality
+grows, most of the points in the hypercube lie outside the inscribed
+hypersphere.
+
+Suppose we have an $N$-dimensional hypercube, with unit-length sides
+centered around the origin $\mathbf{0} = (0, \ldots, 0)$.  The
+hypercube will have $2^N$ corners at the points $\left( \pm
+\frac{1}{2}, \ldots, \pm \frac{1}{2} \right)$.  Because its sides are
+length 1, it will have also have unit hypervolume, because $1^N = 1$.
+
+If $N=1$, the hypercube is a line from $\left( -\frac{1}{2}, \right)$
+to $\left( -\frac{1}{2}, \right)$ of unit length (i.e., the length is
+1).  If $N=2$, the hypercube is a square of unit area with opposite
+corners at $($\left( -\frac{1}{2}, -\frac{1}{2} \right)$ and $\left(
```

Comment:
fixed, not sure how that even parsed.

> On Jun 16, 2016, at 5:45 PM, Daniel Lee <notifications@github.com> wrote:
> 
> In knitr/curse-dims/curse-dims.Rmd:
> 
> > +a cube and a sphere inscribed in it.  When we extend this construction
> > +to higher dimensions, we get hyperspheres inscribed in hypercubes.
> > +This section illustrates the curious fact that as the dimensionality
> > +grows, most of the points in the hypercube lie outside the inscribed
> > +hypersphere.
> > +
> > +Suppose we have an $N$-dimensional hypercube, with unit-length sides
> > +centered around the origin $\mathbf{0} = (0, \ldots, 0)$.  The
> > +hypercube will have $2^N$ corners at the points $\left( \pm
> > +\frac{1}{2}, \ldots, \pm \frac{1}{2} \right)$.  Because its sides are
> > +length 1, it will have also have unit hypervolume, because $1^N = 1$.
> > +
> > +If $N=1$, the hypercube is a line from $\left( -\frac{1}{2}, \right)$
> > +to $\left( -\frac{1}{2}, \right)$ of unit length (i.e., the length is
> > +1).  If $N=2$, the hypercube is a square of unit area with opposite
> > +corners at $($\left( -\frac{1}{2}, -\frac{1}{2} \right)$ and $\left(
> 
> Remove the first $( in this bit of latex: $($\left( -\frac{1}{2}, -\frac{1}{2} \right)$
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>